vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -1163,6 +1164,26 @@ public String getRecommendedDiskController(String guestOsId) throws Exception {
         return null;
     }
 
+    @Override
+    public GuestOsDescriptor getGuestOsDescriptor(String guestOsId) throws Exception {
+        ManagedObjectReference morParent = getParentMor();
+        if (morParent.getType().equals("ClusterComputeResource")) {
```

Comment:
As this validation is used more than once, we could extract it to a method with Javadoc

Reply:
Sorry for getting back this late, I missed these. This "morParent.getType().equals("ClusterComputeResource")" validation is used multiple times but every method has different return values, so I think this is fine